engines/engine.cpp

```diff
@@ -480,10 +482,21 @@ void Engine::checkCD() {
 #endif
 }
 
-bool Engine::shouldPerformAutoSave(int lastSaveTime) {
-	const int diff = _system->getMillis() - lastSaveTime;
-	const int autosavePeriod = ConfMan.getInt("autosave_period");
-	return autosavePeriod != 0 && diff > autosavePeriod * 1000;
+void Engine::handleAutoSave() {
+	const int diff = _system->getMillis() - _lastAutosaveTime;
+
+	if (_autosaveInterval != 0 && diff > (_autosaveInterval * 1000)) {
+		// Save the autosave
+		saveAutosaveIfEnabled();
+	}
+}
+
+void Engine::saveAutosaveIfEnabled() {
+	if (_autosaveInterval != 0 && canSaveAutosaveCurrently())
+		saveGameState(getAutosaveSlot(), _("Autosave"), true);
```

Comment:
Should we warn the player if autosaving failed? With an OSD message perhaps ? At least there should be a warning message in the log/console.

Reply:
Thanks for reminding me. I'd originally decided against showing a message when all I knew was the error message call, but with the OSD display, it'll give the player a non-blocking informative display.